core/src/main/scala/com/rallyhealth/vapors/factfilter/data/FactTable.scala

```diff
@@ -0,0 +1,56 @@
+package com.rallyhealth.vapors.factfilter.data
+
+import com.rallyhealth.vapors.core.data.Indexed
+
+/**
+  * The current state of all the facts in an expression.
+  *
+  * @note some expressions can update the fact table for sub-expressions.
+  */
+// TODO: Use a SortedSet?
+final case class FactTable private (private val facts: Map[String, List[Fact]]) {
+  import cats.syntax.semigroup._
+  import cats.instances.map._
+  import cats.instances.list._
+
+  def add(fact: Fact): FactTable = addAll(fact :: Nil)
+
+  def addAll(facts: List[Fact]): FactTable = {
+    val newFacts = facts.groupBy(_.typeInfo.fullName)
+    new FactTable(this.facts |+| newFacts)
```

Comment:
Can we use method names that are pronounceable, especially when there exists a named `combine` method?